blocks/src/versions/legacy.rs

```diff
@@ -0,0 +1,1692 @@
+        (8, 8) => Some(Block::Water { level: 8 }),
+        (8, 9) => Some(Block::Water { level: 9 }),
+        (8, 10) => Some(Block::Water { level: 10 }),
+        (8, 11) => Some(Block::Water { level: 11 }),
+        (8, 12) => Some(Block::Water { level: 12 }),
+        (8, 13) => Some(Block::Water { level: 13 }),
+        (8, 14) => Some(Block::Water { level: 14 }),
+        (8, 15) => Some(Block::Water { level: 15 }),
+        (9, 0) => Some(Block::Water { level: 0 }),
+        (9, 1) => Some(Block::Water { level: 1 }),
+        (9, 2) => Some(Block::Water { level: 2 }),
+        (9, 3) => Some(Block::Water { level: 3 }),
+        (9, 4) => Some(Block::Water { level: 4 }),
+        (9, 5) => Some(Block::Water { level: 5 }),
+        (9, 6) => Some(Block::Water { level: 6 }),
+        (9, 7) => Some(Block::Water { level: 7 }),
+        (9, 8) => Some(Block::Water { level: 8 }),
+        (9, 9) => Some(Block::Water { level: 9 }),
+        (9, 10) => Some(Block::Water { level: 10 }),
+        (9, 11) => Some(Block::Water { level: 11 }),
+        (9, 12) => Some(Block::Water { level: 12 }),
+        (9, 13) => Some(Block::Water { level: 13 }),
+        (9, 14) => Some(Block::Water { level: 14 }),
+        (9, 15) => Some(Block::Water { level: 15 }),
+        (10, 0) => Some(Block::Lava { level: 0 }),
+        (10, 1) => Some(Block::Lava { level: 1 }),
+        (10, 2) => Some(Block::Lava { level: 2 }),
+        (10, 3) => Some(Block::Lava { level: 3 }),
+        (10, 4) => Some(Block::Lava { level: 4 }),
+        (10, 5) => Some(Block::Lava { level: 5 }),
+        (10, 6) => Some(Block::Lava { level: 6 }),
+        (10, 7) => Some(Block::Lava { level: 7 }),
+        (10, 8) => Some(Block::Lava { level: 8 }),
+        (10, 9) => Some(Block::Lava { level: 9 }),
+        (10, 10) => Some(Block::Lava { level: 10 }),
+        (10, 11) => Some(Block::Lava { level: 11 }),
+        (10, 12) => Some(Block::Lava { level: 12 }),
+        (10, 13) => Some(Block::Lava { level: 13 }),
+        (10, 14) => Some(Block::Lava { level: 14 }),
+        (10, 15) => Some(Block::Lava { level: 15 }),
+        (11, 0) => Some(Block::Lava { level: 0 }),
+        (11, 1) => Some(Block::Lava { level: 1 }),
+        (11, 2) => Some(Block::Lava { level: 2 }),
+        (11, 3) => Some(Block::Lava { level: 3 }),
+        (11, 4) => Some(Block::Lava { level: 4 }),
+        (11, 5) => Some(Block::Lava { level: 5 }),
+        (11, 6) => Some(Block::Lava { level: 6 }),
+        (11, 7) => Some(Block::Lava { level: 7 }),
+        (11, 8) => Some(Block::Lava { level: 8 }),
+        (11, 9) => Some(Block::Lava { level: 9 }),
+        (11, 10) => Some(Block::Lava { level: 10 }),
+        (11, 11) => Some(Block::Lava { level: 11 }),
+        (11, 12) => Some(Block::Lava { level: 12 }),
+        (11, 13) => Some(Block::Lava { level: 13 }),
+        (11, 14) => Some(Block::Lava { level: 14 }),
+        (11, 15) => Some(Block::Lava { level: 15 }),
```

Comment:
couldn't you make this (11, 0..=15)? and then make the level `id & 0xF` LLVM should be able to optimize this out.

Reply:
I've changed this to store `id & 0xF` in a variable before the match block, and then use it in these cases.